src/tests/functional/plugin/shared/src/behavior/infer_request/memory_states.cpp

```diff
@@ -94,7 +94,9 @@ TEST_P(InferRequestVariableStateTest, inferreq_smoke_VariableState_SetState) {
     for (auto&& state : inferReq.QueryState()) {
         state.Reset();
         auto state_val = state.GetState();
-        auto element_count = state_val->size();
+        // We cannot assume state buffer precision is float, for arm64 it maybe fp16.
+        // So the element_cout maybe not the real element cout, it is used to avoid memory out of bound.
+        auto element_count = state_val->byteSize() / sizeof(float);
```

Comment:
Sure we can't make such assumptions. But the precision may be easily checked and a corresponding data type is used. I do understand that this is legacy tests, but, eventually they will be rewritten and it's better to have a proper solution so it will be correctly transferred to the new API.

Reply:
Revert this fixing and create a ticket for state variable behavior:
https://jira.devtools.intel.com/browse/CVS-124395